src/rust_api.rs

```diff
@@ -0,0 +1,1561 @@
+    }
+
+    pub fn mov_sub(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::Sub)
+    }
+
+    pub fn mov_mul(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::Mul)
+    }
+
+    pub fn mov_div(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::Div)
+    }
+
+    pub fn mov_bit_or(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::BitOr)
+    }
+
+    pub fn mov_bit_and(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::BitAnd)
+    }
+
+    pub fn mov_left_shift(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::LShift)
+    }
+
+    pub fn mov_right_shift(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::RShift)
+    }
+
+    pub fn mov_negate(&mut self, arch: Arch) -> Move {
+        self.mov_internal(Source::Imm, arch, OpBits::Negate)
+    }
+
+    pub fn mov_mod(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::Mod)
+    }
+
+    pub fn mov_bit_xor(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::BitXor)
+    }
+
+    pub fn mov(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::NoOp)
+    }
+
+    pub fn mov_signed_right_shift(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::SignRShift)
+    }
+
+    #[inline]
+    fn mov_internal(&mut self, source: Source, arch_bits: Arch, op_bits: OpBits) -> Move {
+        Move {
+            bpf_code: self,
+            src_bit: source,
+            op_bits: op_bits,
+            arch_bits: arch_bits,
+            insn: Insn {
+                opc: 0x00,
+                dst: 0x00,
+                src: 0x00,
+                off: 0x00_00,
+                imm: 0x00_00_00_00
+            }
+        }
+    }
+
+    pub fn swap_bytes(&mut self, endian: Endian) -> SwapBytes {
+        SwapBytes {
+            bpf_code: self,
+            endian: endian,
+            insn: Insn {
+                opc: 0x00,
+                dst: 0x00,
+                src: 0x00,
+                off: 0x00_00,
+                imm: 0x00_00_00_00
+            }
+        }
+    }
+
+    pub fn load(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Undef, 0x00)
+    }
+
+    pub fn load_abs(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Abs, 0x00)
+    }
+
+    pub fn load_ind(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Ind, 0x00)
+    }
+
+    pub fn load_x(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Undef, 0x61)
```

Comment:
Yes, that's `BPF_LDX` here I guess. In `ebpf` module you have for example

    pub const LD_B_REG   : u8 = BPF_LDX   | BPF_MEM | BPF_B;

(for “load byte from register” operation), or from Linux (linux/kernel/bpf/core.c):

    		[BPF_LDX | BPF_MEM | BPF_B] = &&LDX_MEM_B,